### ISSUE TITLE
If dotSecretsStore.p12 gets corrupt, interceptors start to fail 

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19302]
It seems like our secrets store needs to be a little more robust.  In this stack trace, you can see the `EMAWebInterceptor` throw an exception which has the effect of blocking the API call.

This is in edit mode

```
2020-09-22T09:52:58.988-04:00 13:52:58.988 WARN business.ESContentFactoryImpl - ----------------------------------------------
2020-09-22T09:54:53.587-04:00 22-Sep-2020 13:54:53.586 SEVERE [Thread-78] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [RESTAPI] in context with path [] threw exception
2020-09-22T09:54:53.587-04:00 com.dotmarketing.exception.DotRuntimeException: Integrity check failed: java.security.UnrecoverableKeyException: Failed PKCS12 integrity checking
2020-09-22T09:54:53.587-04:00 at com.dotcms.security.apps.SecretsStoreKeyStoreImpl.containsKey(SecretsStoreKeyStoreImpl.java:172)
2020-09-22T09:54:53.587-04:00 at com.dotcms.security.apps.AppsAPIImpl.hasAnySecrets(AppsAPIImpl.java:263)
2020-09-22T09:54:53.587-04:00 at com.dotcms.security.apps.AppsAPIImpl.lambda$8(AppsAPIImpl.java:274)
2020-09-22T09:54:53.587-04:00 at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
2020-09-22T09:54:53.587-04:00 at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
2020-09-22T09:54:53.587-04:00 at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
2020-09-22T09:54:53.587-04:00 at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
2020-09-22T09:54:53.587-04:00 at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
2020-09-22T09:54:53.587-04:00 at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2020-09-22T09:54:53.587-04:00 at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
2020-09-22T09:54:53.587-04:00 at com.dotcms.security.apps.AppsAPIImpl.filterSitesForAppKey(AppsAPIImpl.java:280)
2020-09-22T09:54:53.587-04:00 at com.dotcms.ema.EMAWebInterceptor.existsConfiguration(EMAWebInterceptor.java:167)
2020-09-22T09:54:53.587-04:00 at com.dotcms.ema.EMAWebInterceptor.intercept(EMAWebInterceptor.java:65)
2020-09-22T09:54:53.587-04:00 at com.dotcms.filters.interceptor.SimpleWebInterceptorDelegateImpl.intercept(SimpleWebInterceptorDelegateImpl.java:193)
2020-09-22T09:54:53.587-04:00 at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.runInterceptors(AbstractWebInterceptorSupportFilter.java:38)
2020-09-22T09:54:53.587-04:00 at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:83)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
2020-09-22T09:54:53.587-04:00 at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:73)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:685)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
2020-09-22T09:54:53.587-04:00 at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
2020-09-22T09:54:53.587-04:00 at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800)
2020-09-22T09:54:53.587-04:00 at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
2020-09-22T09:54:53.587-04:00 at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800)
2020-09-22T09:54:53.587-04:00 at org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1699)
2020-09-22T09:54:53.587-04:00 at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
2020-09-22T09:54:53.587-04:00 at org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1050)
2020-09-22T09:54:53.587-04:00 at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:630)
2020-09-22T09:54:53.587-04:00 at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:608)
2020-09-22T09:54:53.587-04:00 at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)
2020-09-22T09:54:53.587-04:00 at sun.nio.ch.Invoker$2.run(Invoker.java:218)
2020-09-22T09:54:53.587-04:00 at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)
```

**Comment by fabrizzio-dotCMS** [comment]
PR: https://github.com/dotCMS/core/pull/19305

**Comment by fabrizzio-dotCMS** [comment]
In order to reproduce this issue, you need to...

1. Save some secrets
2. Update the cluster salt doing something like `update dot_cluster set cluster_salt = 'lol`
3. and then restart tomcat.

**Comment by bryanboza** [comment]
Fixed, tested on release-5.3.9 // Postgres // FF. Tested according the provided information and after the changes I'm unable to reproduce the error
